### PR TITLE
Minor doc updates

### PR DESCRIPTION
**Pull request by 2uasimojo**
One content update for newer `oc adm release extract` syntax.

Added TOCs.

Otherwise mostly typographic stuff.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,24 +2,49 @@
...
+  - [Deploying in cluster](#deploying-in-cluster)
+- [Credentials Requests](#credentials-requests)
+  - [AWS Sample](#aws-sample)
+  - [Azure Sample](#azure-sample)
+- [Instructions to add new cloud provider](#instructions-to-add-new-cloud-provider)
+- [For OpenShift Second Level Operators](#for-openshift-second-level-operators)
+
+## Design Principles
+
+  * The controller should be able to run in either a cluster itself, or in a centralized management cluster, most likely alongside [Hive](https://github.com/openshift/hive).
```
Comment (akhil-rane) [inline README.md:36]: I do not any specific relation between Hive and CCO here. Do we need this in design principles?

File: README.md [hunk 2]
```diff
@@ -221,9 +246,9 @@ OpenStack | N | N | Y | N | N
 oVirt | N | N | Y | N | N
 VMWare | N | N | Y | N | N
```
Comment (akhil-rane) [inline README.md:247]: It would be nice to add alibaba cloud to support matrix here. I think we missed it.

File: README.md [hunk 3]
```diff
@@ -250,29 +275,29 @@ spec:
...
- 2. providerSpec - Contains the [cloud provider specific credentials specification](https://github.com/openshift/cloud-credential-operator/tree/master/pkg/apis/cloudcredential/v1).
+A CredentialRequest spec consists of:
+ 1. `secretRef` - Points to the secret where the credentials should be stored once generated. Can be in a separate namespace from the CredentialsRequest where it can be used by pods. If that namespace does not yet exist, the controller will immediately sync when it is created.
+ 2. `providerSpec` - Contains the [cloud provider specific credentials specification](pkg/apis/cloudcredential/v1).
 
-Once created, assuming admin credentials are available, the controller will provision a user, access key, and user policy in AWS. The access and secret key will be stored in the target secret specified above.
+Once created, assuming admin credentials are available, the controller will provision e.g. a user, access key, and user policy in AWS. The access and secret key will be stored in the target secret specified above.
```
Comment (akhil-rane) [inline README.md:295]: `e.g. ` do we need it?  the following statement looks correct to me
`Once created, assuming admin credentials are available, the controller will provision a user, access key, and user policy in AWS.`

**Review by akhil-rane — COMMENTED** [review]
Looks good!

File: README.md (hunk 3 above)
Comment (2uasimojo) [reply README.md:295]: The way I read it, what happens in AWS is one example of how this works. Against other cloud providers, the specific artifacts provisioned may be different. No?

File: README.md (hunk 2 above)
Comment (2uasimojo) [reply README.md:247]: Sure; I didn't try to change any content with this PR, except in the one place where I knew (by trying it) that the new `oc adm release extract` flags were available.

File: README.md (hunk 3 above)
Comment (akhil-rane) [reply README.md:295]: ahh yes

File: README.md (hunk 2 above)
Comment (akhil-rane) [reply README.md:247]: makes sense. we can keep that for later.

**Comment by akhil-rane** [comment]
Minor corrections and updates to docs. No change in functionality.

/label px-approved
/label qe-approved

**Comment by akhil-rane** [comment]
@jeana-redhat you might want to take a look once and approve it

**Pull request merged**